src/main/java/uk/gov/ch/oauth/Oauth2.java

```diff
@@ -201,4 +202,26 @@ private void regenerateSessionID(HttpServletResponse httpServletResponse) {
 
         httpServletResponse.addCookie(buildSessionCookie(session));
     }
+
+    public void invalidateSession(Session chSession, String signInfo, Store store){
+        final Map<String, Object> sessionData = chSession.getData();
+        if (chSession.getSignInInfo().isSignedIn()){
+            removeSessionInfo(sessionData, signInfo);
+            removeZXSInfo(sessionData, store);
+        }
+    }
+    private void removeSessionInfo(Map<String, Object> sessionData, String signInfo) {
+            final Map<String, Object> signInInfo =
+                    (Map<String, Object>) sessionData.get(signInfo);
+            signInInfo.replace(SessionKeys.SIGNED_IN.getKey(), 1, 0);
+            sessionData.remove(signInfo);
+    }
+    private void removeZXSInfo(Map<String, Object> sessionData, Store store) {
+        final String zxsKey = (String) sessionData.get(".zxs_key");// This is the id of cookie
+        // stored in redis
```

Comment:
Could we change this to a block comment or align them so its more clear that stored in redis  doesn't refer to the next line.

Reply:
I will move the comment onto the one line so that it is clear what line it references.

Reply:
Changed